engine/source/materials/mat/mat036/sigeps36.F

```diff
@@ -700,14 +719,31 @@ SUBROUTINE SIGEPS36(
                 IF (PLAP(I) >= UPARAM(6+J)) JJ(I) = J               
               ENDDO
             ENDDO  
-#include "vectorize.inc" 
-            DO II=1,NINDX                                             
-                  I = INDEX(II) 
-                FAC=UPARAM(6+JJ(I))
-                SRATE(I)=(PLAP(I) - FAC)/(UPARAM(7+JJ(I)) - FAC)
-                YFAC(I,1)=UPARAM(6+NRATE+JJ(I))*FACYLDI(I)
-                YFAC(I,2)=UPARAM(6+NRATE+JJ(I)+1)*FACYLDI(I)
-            ENDDO
+            IF (ISMOOTH == 3) THEN       ! log_n  based interpolation of strain rate
+#include     "vectorize.inc" 
+              DO II=1,NINDX                                             
+                I = INDEX(II) 
+                EPSP1 = MAX(UPARAM(6+JJ(I)), EM20)
+                EPSP2 = UPARAM(7+JJ(I))
+                RFAC(I) = LOG(MAX(PLAP(I),EM20)/EPSP1) / LOG(EPSP2/EPSP1)
+              ENDDO
+            ELSE IF (ISMOOTH == 2) THEN  ! log_10 based interpolation of strain rate
+#include     "vectorize.inc" 
+              DO II=1,NINDX                                             
+                I = INDEX(II) 
+                EPSP1 = MAX(UPARAM(6+JJ(I)), EM20)
+                EPSP2 = UPARAM(7+JJ(I))
+                RFAC(I) = LOG10(MAX(PLAP(I),EM20)/EPSP1) / LOG10(EPSP2/EPSP1)
+              ENDDO
+            ELSE                         ! linear interpolation of strain rate
+#include     "vectorize.inc" 
+              DO II=1,NINDX                                             
+                I = INDEX(II) 
+                EPSP1 = UPARAM(6+JJ(I))
+                EPSP2 = UPARAM(7+JJ(I))
+                RFAC(I) = (PLAP(I) - EPSP1) / (EPSP2 - EPSP1)
```

Comment:
Are we sure that EPSP2 - EPSP1 is not 0, even in signle precision?

Reply:
EPSP1 and EPSP2 are input points of strain rate and they are usually quite different (even in log scale). These are abscissas of  multidimensionnal tables so if they have the same values, the input is simply wrong.

Reply:
And we have a proper starter error instead of a crash in the engine, right?

Reply:
@MaciekWronski , Is it possible to put an error message in this case ?
( in order to avoid possible Nan)

Thanks,
Sébastien

Reply:
You are pointing to the code of linear interpolation which always existed in law 36. It was not modified. I consider it is not the subject of the ticket, but ok, it is always possible to add an error message in case of incorrect input.

Reply:
Actually, there is already an error message in this case in hm_read_mat36.F :
ERROR ID :    478
** INPUT ERROR IN MATERIAL LAW
DESCRIPTION :  
   -- MATERIAL ID: 10
   -- MATERIAL TITLE: Aluminium_example
   STRAIN RATES VALUES MUST BE DEFINED IN ASCENDING ORDER